src/test/java/uk/gov/homeoffice/digital/sas/person/CallistoPersonRestApiIntegrationTest.java

```diff
@@ -0,0 +1,187 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.http.MediaType;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.annotation.DirtiesContext.ClassMode;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.ResultActions;
+import uk.gov.homeoffice.digital.sas.kafka.message.KafkaAction;
+import uk.gov.homeoffice.digital.sas.kafka.message.KafkaEventMessage;
+import uk.gov.homeoffice.digital.sas.person.enums.TermsAndConditions;
+import uk.gov.homeoffice.digital.sas.person.model.Person;
+import uk.gov.homeoffice.digital.sas.person.testutils.KafkaConsumer;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK)
+@DirtiesContext(classMode = ClassMode.AFTER_EACH_TEST_METHOD)
+@AutoConfigureMockMvc
+@TestPropertySource(locations = "classpath:application.properties")
+@ComponentScan("uk.gov.homeoffice.digital.sas.person")
+@EmbeddedKafka(
+    partitions = 1,
+    brokerProperties = {
+        "listeners=PLAINTEXT://localhost:3333",
+        "port=3333"
+    }
+)
+class CallistoPersonRestApiIntegrationTest {
+
+  private static final int CONSUMER_TIMEOUT = 10;
+  private static final String TENANT_ID = "b7e813a2-bb28-11ec-8422-0242ac120002";
+  private static final String TENANT_ID_PARAM = "?tenantId="+TENANT_ID;
+  private static final String PERSON_URL = "/resources/persons";
+  private Person person;
+
+  @Value("${projectVersion}")
+  private String version;
+
+  @Autowired
+  private MockMvc mvc;
+
+  @Autowired
+  private ObjectMapper objectMapper;
+
+  @Autowired
+  private KafkaConsumer kafkaConsumer;
+
+  @BeforeEach
+  void setup() {
+    person = Person.builder()
+        .firstName("John")
+        .lastName("Smith")
+        .version(1)
+        .fteValue(new BigDecimal("1.0000"))
+        .termsAndConditions(TermsAndConditions.MODERNISED)
+        .build();
+  }
+
+  @Test
+  void shouldReturnEmptyListOfItemsWhenNoPersonDataPresent() throws Exception {
+
+    mvc.perform(get(PERSON_URL+ TENANT_ID_PARAM)
+            .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(status().isOk())
+        .andExpect(content()
+            .contentTypeCompatibleWith(MediaType.APPLICATION_JSON))
+        .andExpect(jsonPath("$.items", empty()));
```

Comment:
It feels like there should be an assertion before this one to assert that there actually isn't any person data in the db.
You might be able to just autowire in the repo and check it's empty.

Reply:
I don't see the need for the additional assertion. H2 database should be reset before each test thanks to `@DirtiesContext(classMode = ClassMode.AFTER_EACH_TEST_METHOD)`

Reply:
Ah yes I forgot that annotation was present. You're right, in this context the DB would always be clear.